### PR TITLE
DOC: update references to pyansys/actions

### PR DESCRIPTION
**Pull request by jorgepiloto**
This pull-request adds documentation about pyansys/actions and updates all the CI/CD references to those.

Lots of lines simplified now that the CI/CD workflows snippets are using the pyansys/actions.

### CODE REVIEW

File: doc/source/how-to/code/docs.yml
Comment (MaxJPRey) [inline doc/source/how-to/code/docs.yml:19]: That is also a good way to expose/remind the convention for the doc URLs.

Comment (MaxJPRey) [inline doc/source/how-to/code/docs.yml:19]: Just to add the `dev` prefix.
```suggestion
        cname: "dev.<library>.docs.pyansys.com"
```

File: doc/source/how-to/code/docs.yml
Comment (MaxJPRey) [inline doc/source/how-to/code/docs.yml:33]: Do we really need it?

File: doc/source/how-to/code/docs.yml
Comment (jorgepiloto) [reply doc/source/how-to/code/docs.yml:19]: Thanks for suggesting this, Maxime. The `pyansys/actions/doc-deploy-dev` assume that all content is pushed to the `gh-pages` of the repository, and thus, that the CNAME is the same for both stable and development documentation.

This action is linked to the multi-version documentation, so not need to add the `dev` at the start of the CNAME.

File: doc/source/how-to/code/docs.yml
Comment (jorgepiloto) [reply doc/source/how-to/code/docs.yml:33]: The script in charge of dealing with the multi-version doc is written in Python. It makes use of the standard library and does not require any dependency. I just declared this as input parameter in case users would like to specify the Python version used to run the script.

Let me delete line 33. No need to pass as Python 3.10 is used if no version is provided.

File: doc/source/how-to/code/docs.yml
Comment (RobPasMue) [reply doc/source/how-to/code/docs.yml]: Yep, agreed! We should help other repos migrate to this multi-versioning mechanism. It will imply:
* Closing documentation-specific repositories: ``py*-dev-docs`` and ``py*-docs``
* Deleting the ``dev.*.docs.pyansys.com`` domains from azure
* Implementing the ``gh-pages`` branch in each of the repositories and helping them update their ``conf.py``

A lot of work coming up in general, but it's going to be super useful =)

File: .github/workflows/ci-build.yml
Comment (RobPasMue) [inline .github/workflows/ci-build.yml:64]:
```suggestion
  release-github:
```

File: .github/workflows/ci-build.yml
Comment (RobPasMue) [inline .github/workflows/ci-build.yml:93]: Don't really know if zipping the pdf file is really necessary. I would only zip the HTML docs (since it's composed of multiple files)- But the PDF docs are just one.

File: doc/source/how-to/documenting.rst
Comment (RobPasMue) [inline doc/source/how-to/documenting.rst:504]: Here you are pointing at ``@v1``  while in the ``yml`` files you point at ``@main``. I do think it's better to point at a stable version rather than main for these workflows. Let dependabot update them if they ever copypaste from here =)

This will imply changing all ``yml`` files.

File: doc/source/how-to/documenting.rst
Comment (RobPasMue) [inline doc/source/how-to/documenting.rst:533]:
```suggestion
If you are planning to deploy documentation to a repository other than
```

File: doc/source/how-to/packaging.rst
Comment (RobPasMue) [inline doc/source/how-to/packaging.rst:343]: Should we talk as well about allowing dependabot to update your actions? I can't seem to find this info anywhere in the dev-guide, but I may be mistaken.

File: .github/workflows/ci-build.yml
Comment (jorgepiloto) [reply .github/workflows/ci-build.yml:64]: Good catch!

File: .github/workflows/ci-build.yml
Comment (jorgepiloto) [reply .github/workflows/ci-build.yml:93]: Makes sense! Fixing this... ⚙️

File: doc/source/how-to/code/docs.yml
Comment (jorgepiloto) [reply doc/source/how-to/code/docs.yml]: Let me add these lines also to the dev-guide, so we do not forget about how to perform the migration in our libraries.

File: .github/workflows/ci-build.yml
Comment (jorgepiloto) [reply .github/workflows/ci-build.yml]: Solved in f19aee5.

File: .github/workflows/ci-build.yml
Comment (Revathyvenugopal162) [inline .github/workflows/ci-build.yml:31]:
```suggestion
      - name: "documentation style checks"
```

Comment (jorgepiloto) [reply .github/workflows/ci-build.yml:31]: Good catch, @Revathyvenugopal162!

File: doc/source/how-to/documenting.rst
Comment (Revathyvenugopal162) [inline doc/source/how-to/documenting.rst:540]:
```suggestion
    translates to more maintenance work and does not support the multi-version
```

File: doc/source/how-to/documenting.rst
Comment (jorgepiloto) [reply doc/source/how-to/documenting.rst:504]: Sure, let me make sure that everything points to the `@v1`. This may be a result of copy-pasting from the development documentation of the actions.

Comment (jorgepiloto) [reply doc/source/how-to/documenting.rst:504]: Solved!

File: doc/source/how-to/documenting.rst
Comment (Revathyvenugopal162) [inline doc/source/how-to/documenting.rst:576]: it will be available on if you setup the ``"navbar_end": ["version-switcher", "theme-switcher", "navbar-icon-links"],`` in conf file otherwise it will be on top left of the page.

Comment (RobPasMue) [reply doc/source/how-to/documenting.rst:576]: Can't we force all this in the ansys-sphinx-theme?

Comment (Revathyvenugopal162) [reply doc/source/how-to/documenting.rst:576]: yes. i was looking that right now. we can do that.

Comment (jorgepiloto) [reply doc/source/how-to/documenting.rst:576]: If we could force this directly from our theme it would be awesome!

Comment (RobPasMue) [reply doc/source/how-to/documenting.rst:576]: Totally... I mean, I'm fine adding it in one repo.. but if we have to add it in 3 or more... please add it to the main theme 😄

File: doc/source/how-to/code/docs.yml
Comment (jorgepiloto) [reply doc/source/how-to/code/docs.yml]: Alright! I just added more information on how to perform the migration. The section includes common steps and mature projects indications.

File: doc/source/how-to/packaging.rst
Comment (jorgepiloto) [reply doc/source/how-to/packaging.rst]: I briefly cited the capabilities of dependabot for updating actions versions. On top of that, code snippets were updated with the required code for enabling such feature.

**Review by MaxJPRey — APPROVED** [review]
LGTM.

**Comment by RobPasMue** [comment]
Ready to merge right @jorgepiloto ?

**Comment by jorgepiloto** [comment]
That's right, @RobPasMue. I would like to have a review by @PipKat, just in case.

File: .github/workflows/ci-build.yml
Comment (PipKat) [inline .github/workflows/ci-build.yml:54]: @jorgepiloto Can we just make it "Deploy developer documentation"? Technically, we'd want developer's (singular, possessive, but I don't think we need or want apostrophes in here. Although after reading the names of others, maybe we do just put the apostrophe because they are more verbose than I expected. It's up to you.

File: doc/source/how-to/code/docs.yml
Comment (PipKat) [inline doc/source/how-to/code/docs.yml:5]: "Run action for building documentation" seems sufficient. If you want Ansys in there before the word documentation, that'd be OK.

File: doc/source/how-to/code/docs.yml
Comment (PipKat) [inline doc/source/how-to/code/docs.yml:11]: "Deploy developer documentation" here to match the earlier suggestion?

File: doc/source/how-to/code/docs.yml
Comment (PipKat) [inline doc/source/how-to/code/docs.yml:16]: Is this the dev version of the documentation and the next section the release version of the documentation. If so, I think "latest" is OK, but "stable" isn't very descriptive.

File: doc/source/how-to/code/release.yml
Comment (PipKat) [inline doc/source/how-to/code/release.yml:15]: Should we make this parallel to the previous name, perhaps "Release to the public PyPI repository"? What is the difference between line 15 and line 19--because now I see that my suggestion is exactly what appears in line 19.

File: doc/source/how-to/code/style.yml
Comment (PipKat) [inline doc/source/how-to/code/style.yml:12]:
```suggestion
    - name: "Run PyAnsys documentation style checks"
```

File: doc/source/how-to/code/tests.yml
Comment (PipKat) [inline doc/source/how-to/code/tests.yml:16]:
```suggestion
    - name: "Upload coverage results"
```

Comment (PipKat) [reply doc/source/how-to/code/tests.yml:16]: I know the report shown is in initial caps--but here, I'd use lower case to be consistent with other names.

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:478]:
```suggestion
   library. To enable support for multi-version documentation, see
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:487]:
```suggestion
your documentation build on GitHub and deployed either on release or on a push to
the ``main`` branch. You can do this via `GitHub Actions`_ by taking advantage of
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:498]: Of course, if you make my suggested name changes from earlier, they will also need to be made in the subsequent lines.

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:541]:
```suggestion
    would require more maintenance and would not support multi-version
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:544]:
```suggestion
To deploy the documentation to another repository, use this workflow:
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:576]:
```suggestion
- Create also a ``.nojekyll`` file that tells GitHub about the nature of the website.
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:577]:
```suggestion
- Create a ``release/`` directory with a ``versions.json`` file inside of it.
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:579]:
```suggestion
The ``version.json`` file contains this content:
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:593]: I'm not the best at reading between deleted and added lines. However, I don't see new lines to replace lines 680-686. I only see the tagging for a code block. Does "import os" to be moved prior to the "Additionally, you must...."?

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:605]:
```suggestion
        """Evaluate the version match for the multi-version documentation."""
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:622]:
```suggestion
- Send an e-mail to ``support@pyansys.com`` if you need to remove your old development ``CNAME``.
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:628]:
```suggestion
Once you complete the configuration changes, your project is ready to use multi-version
documentation in an automated way. This means that every time you release a
new version, it is added to the drop-down button in the documentation page
of the project.
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:633]:
```suggestion
    By default, the documentation drop-down shows only the development version and
    the latest three stable versions.
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:637]:
```suggestion
Mature project configuration
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:642]:
```suggestion
A *mature* project is a library that already has  several released
versions. Developers of a mature project should apply previous changes
in the ``main`` branch of their project and then *cherry-pick* these changes in
the latest three stable release branches. This ensures that the multi-version
documentation is able to function as desired.
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:646]:
```suggestion
Because new ``release/X.Y`` branches are created from ``main`` branch with
the new configuration, no additional changes are required for future releases
of your project.
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:652]:
```suggestion
    If you require support when migrating to multi-version documentation,
    send an email to ``support@pyansys.com``.
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:659]:
```suggestion
constructed using this structure:
```

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:671]:
```suggestion
    PyAnsys projects support multi-version documentation, meaning that
    stable and development versions are collected under the same webpage. A
    drop-down button for selecting the desired version should be available in the
    top right corner of the navigation bar on the documentation page.
```

Comment (PipKat) [reply doc/source/how-to/documenting.rst:671]: Google style guide says documentation should be timeless, so no "now" or "in the past" or "in the future" stuff.

File: doc/source/how-to/documenting.rst
Comment (PipKat) [inline doc/source/how-to/documenting.rst:676]:
```suggestion
- The documentation URL for the latest stable release is `<https://aedt.docs.pyansys.com/>`_.
- The documentation URL for the latest development version is `<https://dev.aedt.docs.pyansys.com/>`_.
```

File: doc/source/how-to/grpc-api-packages.rst
Comment (PipKat) [inline doc/source/how-to/grpc-api-packages.rst:250]:
```suggestion
dependencies, see :ref:`Required Dependencies`.
```

Comment (PipKat) [reply doc/source/how-to/grpc-api-packages.rst:250]: The title should be sentence case, so "Required dependenices". However, I didn't want to break the link.

File: doc/source/how-to/packaging.rst
Comment (PipKat) [inline doc/source/how-to/packaging.rst:339]:
```suggestion
configuration file into your repository. In this file, you should specify the
```

File: doc/source/how-to/packaging.rst
Comment (PipKat) [inline doc/source/how-to/packaging.rst:340]:
```suggestion
location of the project's requirement files so that Dependabot knows where to
```

File: doc/source/how-to/packaging.rst
Comment (PipKat) [inline doc/source/how-to/packaging.rst:341]:
```suggestion
look. On top of that, Dependabot is capable of updating ``GitHub actions``
```

File: doc/source/how-to/packaging.rst
Comment (PipKat) [inline doc/source/how-to/packaging.rst:342]: Does GitHub actions need to be a literal string. It seems a common term and not something that is code-specific. Also, I'd recommend "...capable of updating versions for GitHub actions."

File: doc/source/how-to/packaging.rst
Comment (PipKat) [inline doc/source/how-to/packaging.rst:448]:
```suggestion
T enable security updates and notifications for your repository, go to
```

File: doc/source/how-to/testing.rst
Comment (PipKat) [inline doc/source/how-to/testing.rst:524]:
```suggestion
unfamiliar with GitHub actions, see `GitHub Actions`_ for an overview.
```

**Review by PipKat — APPROVED** [review]

File: doc/source/how-to/packaging.rst
Comment (jorgepiloto) [reply doc/source/how-to/packaging.rst:342]: You are right Kathy. No need to have this sort of highlight on the GitHub actions word.

File: .github/workflows/ci-build.yml
Comment (jorgepiloto) [reply .github/workflows/ci-build.yml:54]: Ops... This shall read "Deploy development documentation".

File: doc/source/how-to/code/docs.yml
Comment (jorgepiloto) [reply doc/source/how-to/code/docs.yml:5]: I just modified this name as "Build project documentation". It is shorter and more explanatory.

File: doc/source/how-to/code/release.yml
Comment (jorgepiloto) [reply doc/source/how-to/code/release.yml:15]: The first name refers to the job name while the second is a step name. Since there is only one step in this job, I think we could even get rid of the step name... Anyway, I agree on using the same name for both.

**Comment by RobPasMue** [comment]
Now that @PipKat's review is finished and we all have approved I am merging this PR =)

**Pull request merged**
